### PR TITLE
Modernize dummy app

### PR DESCRIPTION
**Pull request by marcoow**
This modernizes the dummy app by:

* using `async`/`await` instead of dealing with promises
* using closure actions instead of old actions or `sendAction`
* using ember-data's package imports
* using native inputs in the login form instead of the `input` helper

It also fixes handling of invalid credentials and other login errors in the `<LoginForm>` which is currently broken.

### CODE REVIEW

File: tests/dummy/app/templates/components/login-form.hbs [hunk 1]
```diff
@@ -7,14 +7,14 @@
 <form {{action 'authenticateWithOAuth2' on='submit'}}>
   <div class="form-group">
     <label for="identification">Login</label>
-    {{input value=identification placeholder='Enter Login' class='form-control'}}
+    <input type="text" placeholder="Enter Login" class="form-control" onchange={{action 'updateIdentification'}} value={{this.identification}} />
```
Comment (Turbo87) [inline tests/dummy/app/templates/components/login-form.hbs]: hmm, what's the reason for this change? why not keep using the `<Input>` component?

File: tests/dummy/app/components/login-form.js [hunk 2]
```diff
@@ -6,15 +6,18 @@ export default Component.extend({
...
+      try {
+        let { identification, password } = this;
+        await this.get('session').authenticate('authenticator:oauth2', identification, password);
+
+        if (this.rememberMe) {
+          this.get('session').set('store.cookieExpirationTime', 60 * 60 * 24 * 14);
+        }
+      } catch (response) {
+        let responseBody = await response.clone().json();
+        this.errorMessage = responseBody.error;
```
Comment (Turbo87) [inline tests/dummy/app/components/login-form.js]: this looks quite different than the previous implementation 🤔

File: tests/dummy/app/services/session-account.js [hunk 3]
```diff
@@ -1,22 +1,14 @@
...
-        }, reject);
-      } else {
-        resolve();
-      }
-    });
+  async loadCurrentUser() {
+    let accountId = this.get('session.data.authenticated.account_id');
+    if (accountId) {
+      let account = await this.get('store').find('account', accountId);
+      this.account = account;
```
Comment (Turbo87) [inline tests/dummy/app/services/session-account.js]: should we not keep using `this.set()` here?

Comment (marcoow) [reply tests/dummy/app/services/session-account.js]: yes, fixed

File: tests/dummy/app/templates/components/login-form.hbs (hunk 1 above)
Comment (marcoow) [reply tests/dummy/app/templates/components/login-form.hbs]: I find using native inputs easier to follow and more explicit. What would be an advantage of the `<Input>` component?

File: tests/dummy/app/components/login-form.js (hunk 2 above)
Comment (marcoow) [reply tests/dummy/app/components/login-form.js]: Yes, this is actually broken on `master` currently…

Comment (Turbo87) [reply tests/dummy/app/components/login-form.js]: then shouldn't we fix this in a dedicated PR?

I find this PR somewhat hard to review because it's changing so many things at the same time

Comment (marcoow) [reply tests/dummy/app/components/login-form.js]: Yes, makes sense: #2138

File: tests/dummy/app/templates/components/login-form.hbs [hunk 4]
```diff
@@ -4,17 +4,17 @@
 <a {{action 'authenticateWithGoogleImplicitGrant'}} class="btn btn-primary" href="#" role="button">Login with Google (Implicit Grant)</a>
 <hr/>
 <p>or login with OAuth 2.0 <em>Resource Owner Password Credentials</em>:</p>
-<form {{action 'authenticateWithOAuth2' on='submit'}}>
+<form onsubmit={{action 'authenticateWithOAuth2'}}>
```
[the remark this answers is not in the document]
Comment (marcoow) [reply tests/dummy/app/templates/components/login-form.hbs]: ok, can revert
